src/huggingface_hub/repocard.py

```diff
@@ -1,30 +1,288 @@
+            data_dict = {}
+            self.text = content
+
+        model_index = data_dict.pop("model-index", None)
+        if model_index:
+            try:
+                model_name, eval_results = model_index_to_eval_results(model_index)
+                data_dict["model_name"] = model_name
+                data_dict["eval_results"] = eval_results
+            except KeyError:
+                logger.warning(
+                    "Invalid model-index. Not loading eval results into CardData."
+                )
+
+        self.data = CardData(**data_dict)
+
+    def __str__(self):
+        return f"---\n{self.data.to_yaml()}\n---\n{self.text}"
+
+    def save(self, filepath: Union[Path, str]):
+        r"""Save a RepoCard to a file.
+
+        Args:
+            filepath (`Union[Path, str]`): Filepath to the markdown file to save.
+
+        Example:
+            >>> from huggingface_hub import RepoCard
+            >>> card = RepoCard("---\nlanguage: en\n---\n# This is a test repo card")
+            >>> card.save("/tmp/test.md")
+        """
+        filepath = Path(filepath)
+        filepath.parent.mkdir(parents=True, exist_ok=True)
+        filepath.write_text(str(self))
+
+    @classmethod
+    def load(cls, repo_id_or_path: Union[str, Path], repo_type=None, token=None):
+        """Initialize a RepoCard from a Hugging Face Hub repo's README.md or a local filepath.
+
+        Args:
+            repo_id_or_path (`Union[str, Path]`):
+                The repo ID associated with a Hugging Face Hub repo or a local filepath.
+            repo_type (`str`, *optional*):
+                The type of Hugging Face repo to push to. Defaults to None, which will use
+                use "model". Other options are "dataset" and "space".
+            token (`str`, *optional*):
+                Authentication token, obtained with `huggingface_hub.HfApi.login` method. Will default to
+                the stored token.
+
+        Returns:
+            `huggingface_hub.RepoCard`: The RepoCard (or subclass) initialized from the repo's
+                README.md file or filepath.
+
+        Example:
+            >>> from huggingface_hub import RepoCard
+            >>> card = RepoCard.load("nateraw/food")
+            >>> assert card.data.tags == ["generated_from_trainer", "image-classification", "pytorch"]
+        """
+
+        if Path(repo_id_or_path).exists():
+            card_path = Path(repo_id_or_path)
+        else:
+            card_path = hf_hub_download(
+                repo_id_or_path,
+                REPOCARD_NAME,
+                repo_type=repo_type,
+                use_auth_token=token,
+            )
+
+        return cls(Path(card_path).read_text())
+
+    def validate(self, repo_type="model"):
+        """Validates card against Hugging Face Hub's model card validation logic.
+        Using this function requires access to the internet, so it is only called
+        internally by `huggingface_hub.ModelCard.push_to_hub`.
```

Comment:
Had this discussion w/ Adrin in original PR in `modelcards`, we opted to have it not return anything. It errors out if theres an issue, so that's how we're communicating it vs. returning `True`/`False`.